src/App/Fossa/Analyze.hs

```diff
@@ -51,18 +55,50 @@ import qualified Strategy.Python.ReqTxt as ReqTxt
 import qualified Strategy.Python.SetupPy as SetupPy
 import qualified Strategy.Ruby.BundleShow as BundleShow
 import qualified Strategy.Ruby.GemfileLock as GemfileLock
+import System.Exit (die)
 import Types
 import OptionExtensions
 
 data ScanDestination
   = UploadScan UrlOption Text ProjectMetadata -- ^ upload to fossa with provided api key and base url
   | OutputStdout
   deriving (Generic)
- 
-analyzeMain :: Severity -> ScanDestination -> Maybe Text -> Maybe Text -> IO ()
-analyzeMain logSeverity destination name revision = do
-  basedir <- getCurrentDir
-  withLogger logSeverity $ analyze basedir destination name revision
+
+data AnalyzeOptions = AnalyzeOptions
+  { analyzeOutput :: Bool
+  , analyzeMetadata :: ProjectMetadata
+  , analyzeBaseDir :: FilePath
+  }
+
+analyzeOpts :: Parser AnalyzeOptions
+analyzeOpts =
+  AnalyzeOptions
+    <$> switch (long "output" <> short 'o' <> help "Output results to stdout instead of uploading to fossa")
+    <*> metadataOpts
+    <*> baseDirArg
+
+metadataOpts :: Parser ProjectMetadata
+metadataOpts =
+  ProjectMetadata
+    <$> optional (strOption (long "title" <> help "the title of the FOSSA project. (default: the project name)"))
+    <*> optional (strOption (long "branch" <> help "this repository's current branch (default: current VCS branch)"))
+    <*> optional (strOption (long "project-url" <> help "this repository's home page"))
+    <*> optional (strOption (long "jira-project-key" <> help "this repository's JIRA project key"))
+    <*> optional (strOption (long "link" <> help "a link to attach to the current build"))
+    <*> optional (strOption (long "team" <> help "this repository's team inside your organization"))
+    <*> optional (strOption (long "policy" <> help "the policy to assign to this project in FOSSA"))
+
+mkScanDestination :: Bool -> Maybe Text -> UrlOption -> ProjectMetadata -> IO ScanDestination
```

Comment:
This type signature is very vague to me. What's the `Bool`? what's the `Maybe Text`? There are two ways to address this -- one is to use a newtype wrapper for API key (which I should've done when I first wrote this, and is kind of a pervasive issue across these modules) + a clearer datatype than `Bool`.

The other way is to name the arguments and use `case`:

```hs
mkScanDestination forceStdout apiKey baseUrl metadata =
  case (forceStdout, apiKey) of
    (True, _) -> ...
    (_, Nothing) -> die ...
    (_, Just key) -> ...
```

Reply:
Agree with your point but not sure how to proceed.  Let's pair on this.